### PR TITLE
Most all my new FFT stuff in a decent state

### PR DESCRIPTION
**Pull request by romainlebreton**
It should not impact any other tests but let's still do a pull request.
I will still have to split bench-new-fft in a test and a bench file.

### CODE REVIEW

**Comment by ClementPernet** [comment]
trigger CI

**Comment by ClementPernet** [comment]
trigger CI

**Comment by ClementPernet** [comment]
trigger CI

File: examples/bench-fft.C [hunk 1]
```diff
@@ -133,7 +133,7 @@ bool check_DIF(const Field& fld, size_t kmax, long seed) {
 		passed &= DFT_sanity_check(MulDom,&FFT_t::FFT_DIT_Harvey_mod4p_iterative3x3,x,y, "DIT_Harvey_mod4p_iterative3x3");
 		// check 4x1 SSE		
 		//passed &= DFT_sanity_check(MulDom,&FFT_t::FFT_DIT_Harvey_mod4p_iterative4x1_SSE,x,y, "DIT_Harvey_mod4p_iterative4x1_SSE");
-#ifdef __LINBOX_USE_AVX2
+#ifdef __LINBOX_HAVE_AVX_INSTRUCTIONS2
```
Comment (ClementPernet) [inline examples/bench-fft.C]: test diff comment

**Comment by ClementPernet** [comment]
trigger CI

**Comment by ClementPernet** [comment]
trigger CI

**Comment by ClementPernet** [comment]
dqfsqdf

**Comment by ClementPernet** [comment]
Note that the build status is marked as failed due to some unexpected failure in jenkins. I reran the jenkins build and it now terminates with success. 
https://ci.inria.fr/linbox/view/LinBox%20ecosystem/job/LinBox-Pull-Requests/7/
However I can't get the status of this PR to be updated accordingly.

**Comment by romainlebreton** [comment]
@ClementPernet I have no idea why I have [this error ](https://ci.inria.fr/linbox/job/LinBox-Pull-Requests/8/Archi=linbox-ubuntu-i386-icpc-clang,Compiler=icpc,NTL=withoutNTL/parsed_console/) on one particular computer. For me, the code is OK and should be merged.

**Comment by ClementPernet** [comment]
Ok. It's again some issue with partially installed dependencies (too many concurrent jobs running on the CI).
I'll make sur to test it again, once I'm done with fixing fflas, and see if I can merge before the release.

**Comment by ClementPernet** [comment]
example directory is supposed to show simple files illustrating how to use one function, mostly a solution.
The goal is pedagogical, showing simple usecases, and also to provide one with executables running each of the solution function on a matrix read from a file.
You are proposing here benchmarks in the example dir. They should be placed in the benchmarks directory and conform to the std output of benchmarks.

How important is it for you to have the whole PR merged in the new release?
If it is, then we can merge it as is, as the example dir is already a mess,  but you'll need to sort this thing out later on.

**Comment by romainlebreton** [comment]
Yes, I know about tests, bench and examples directory and I should definitely do it !
This merge request is not crucial but I would still do it because it corrects many stuffs on files that are already present on the master branch.

**Comment by ClementPernet** [comment]
trigger CI

**Comment by ClementPernet** [comment]
@romainlebreton : please wait until the CI has finished and reports no error before merging a PR.
I'm reverting it so we get the notification of success or failure

**Comment by romainlebreton** [comment]
I merged it because the last commit (merge) was not introducing anything new.

**Pull request merged**